ebos/ecltransmissibility.hh

```diff
@@ -859,22 +899,28 @@ private:
         // cells merged as an result of minpv.
         const auto& eclState = vanguard_.eclState();
         const auto& eclGrid = eclState.getInputGrid();
-
-        const std::vector<double>& ntg =
-            eclState.get3DProperties().getDoubleGridProperty("NTG").getData();
-
-        averageNtg = ntg;
-
         bool opmfil = eclGrid.getMinpvMode() == Opm::MinpvMode::OpmFIL;
+        std::vector<double> ntg;
 
+#ifdef ENABLE_3DPROPS_TESTING
+        ntg = eclState.fieldProps().get_global_double("NTG");
+#else
+        ntg = eclState.get3DProperties().getDoubleGridProperty("NTG").getData();
+#endif
         // just return the unmodified ntg if opmfil is not used
+        averageNtg = ntg;
         if (!opmfil)
             return;
 
-        const auto& porv = eclState.get3DProperties().getDoubleGridProperty("PORV").getData();
-        const auto& actnum = eclState.get3DProperties().getIntGridProperty("ACTNUM").getData();
         const auto& cartMapper = vanguard_.cartesianIndexMapper();
         const auto& cartDims = cartMapper.cartesianDimensions();
+#ifdef ENABLE_3DPROPS_TESTING
+        const auto& actnum = eclState.fieldProps().actnum();
+        const auto& porv = eclState.fieldProps().porv(true);
```

Comment:
Confusing that actnum and porv are vectors of cartesian size. How can a developer know which properties have which size?

Reply:
For someone with an unhealthy interest in Eclipse file formats this is very natural, but I see your point. The `ACTNUM` vector is a uncompressed vector of `nx*ny*nz` elements of 0 and 1; where 0 denotes an inactive cell and 1 denotes an active cell - to actually fill this role it must be of full length. I am not arguing this representation is the best way to identify active and inactive cells, but it is way it is done in both the input and output files - and familiar to developers who have worked with these files. 

The `PORV` vector is of course mainly the porevolume of the respective cells, but it also serves as a "floating point ACTNUM vector". The `INIT` file contains many properties like `PERMX, PORO` and `SWL` which are all represented as compressed vectors; and in addition the `INIT` file contains the `PORV` vector as an uncompressed vector. In the uncompressed PORV vector elements with value zero identify inactive cells; so in the output domain inspecting `PORV` from the `INIT` file is an alternative to inspecting `ACTNUM` from the `EGRID` file.

In this code the `ACTNUM` vector is *always* uncompressed; but for `PORV` you can adjust it with the boolean flag.

Reply:
Maybe we should put this somewhere (e.g. wiki) for reference for people who try to live healthy :laughing: